File: python/TwoSum.py
from typing import List


class TwoSum:
        def solution(self, nums: List[int], target: int) -> List[int]:
                num_map = {}
                for i, num in enumerate(nums):
                        remainingNum = target - num
                        if remainingNum in num_map:
                                return [num_map[remainingNum], i]
                        num_map[num] = i
                return []


if __name__ == "__main__":
        nums = [12, 7, 11, 15]
        target = 23
        two_sum = TwoSum()
        result = two_sum.solution(nums, target)
        print(result)

File: java/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
        public int[] solution(int[] nums, int target) {
                Map<Integer, Integer> numMap = new HashMap<>();
                for (int i = 0; i < nums.length; i++) {
                        int remainingNum = target - nums[i];
                        if (numMap.containsKey(remainingNum)) {
                                return new int[] {numMap.get(remainingNum), i};
                        }
                        numMap.put(nums[i], i);
                }
                return new int[] {};
        }

        public static void main(String[] args) {
                TwoSum twoSum = new TwoSum();
                int[] nums = {2, 7, 11, 15};
                int target = 9;
                int[] result = twoSum.solution(nums, target);
                System.out.println("[" + result[0] + ", " + result[1] + "]");
        }
}